app/components/CreateCrashRecordModal.tsx

```diff
@@ -0,0 +1,377 @@
+   */
+  show: boolean;
+}
+
+type UnitInputs = {
+  unit_desc_id: string;
+  fatality_count: number;
+  injury_count: number;
+};
+
+type CrashInputs = {
+  case_id: string;
+  crash_timestamp: string;
+  created_by: string;
+  cris_schema_version: "2023";
+  is_temp_record: boolean;
+  private_dr_fl: boolean;
+  rpt_city_id: number;
+  rpt_sec_street_name: string;
+  rpt_street_name: string;
+  units_cris: UnitInputs[];
+  updated_by: string;
+};
+
+const DEFAULT_UNIT: UnitInputs = {
+  unit_desc_id: "",
+  fatality_count: 0,
+  injury_count: 0,
+};
+
+const DEFAULT_FORM_VALUES: CrashInputs = {
+  case_id: "",
+  crash_timestamp: "",
+  created_by: "",
+  cris_schema_version: "2023",
+  is_temp_record: true,
+  private_dr_fl: false,
+  rpt_city_id: 22,
+  rpt_sec_street_name: "",
+  rpt_street_name: "",
+  units_cris: [{ ...DEFAULT_UNIT }],
+  updated_by: "",
+};
+
+/**
+ * Construct a unit record with people records such
+ * that it can be included in a crash insert mutation
+ */
+const makeUnitRecord = (unit: UnitInputs, index: number, userEmail: string) => {
+  const unitNumber = index + 1;
+
+  const unitRecord: Record<string, unknown> = {
+    unit_nbr: unitNumber,
+    unit_desc_id: unit.unit_desc_id,
+    cris_schema_version: "2023",
+    updated_by: userEmail,
+    created_by: userEmail,
+  };
+
+  const people = [];
+
+  // create fatal injuries
+  for (let i = 0; i < Number(unit.fatality_count); i++) {
+    const personRecord = {
+      prsn_nbr: i + 1,
+      unit_nbr: unitNumber,
+      prsn_injry_sev_id: 4,
+      cris_schema_version: "2023",
+      is_primary_person: true,
+      updated_by: userEmail,
+      created_by: userEmail,
+    };
+
+    people.push(personRecord);
+  }
+
+  // create other injuries
+  for (let i = 0; i < Number(unit.injury_count); i++) {
+    people.push({
+      prsn_nbr: unit.fatality_count + i + 1,
+      unit_nbr: unitNumber,
+      prsn_injry_sev_id: 1,
+      cris_schema_version: "2023",
+      is_primary_person: true,
+      updated_by: userEmail,
+      created_by: userEmail,
+    });
+  }
+
+  return { ...unitRecord, people_cris: { data: people } };
+};
+
+/**
+ * Modal form component used for creating a temporary crash record
+ */
+export default function CreateCrashRecordModal({
+  onClose,
+  onSubmitCallback,
+  show,
+}: CreateCrashModalProps) {
+  const { user } = useAuth0();
+  const { data: unitTypes, isLoading: isLoadingUnitTypes } =
+    useQuery<LookupTableOption>({
+      query: UNIT_TYPES_QUERY,
+      typename: "lookups_unit_desc",
+    });
+  const { mutate, loading: isSubmitting } = useMutation(CREATE_CRASH);
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    reset,
+  } = useForm<CrashInputs>({
+    defaultValues: {
+      ...DEFAULT_FORM_VALUES,
+      created_by: user?.email || "",
+      updated_by: user?.email || "",
+    },
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "units_cris",
+  });
```

Comment:
this is a neat react-hook-form feature that enables repeatable arrays of form groups—exactly what we need for the **Units** section of this form

Reply:
oooh this is a TIL! I was wondering how this was going to work when I was testing. 🚀